javabuilder-authorizer/http_authorizer_function.rb

```diff
@@ -15,5 +18,45 @@ def lambda_handler(event:, context:)
 
   decoded_token = JwtHelper.decode_token(jwt_token, origin)
   return JwtHelper.generate_deny(route_arn) unless decoded_token
-  return JwtHelper.generate_allow(route_arn, decoded_token)
+
+  token_payload = decoded_token[0]
+  token_status = get_token_status(context, token_payload['sid'])
+  return JwtHelper.generate_deny(route_arn) unless token_status == TokenStatus::VALID_HTTP
+
+  JwtHelper.generate_allow(route_arn, token_payload)
+end
+
+def get_token_status(context, sid)
+  client = Aws::DynamoDB::Client.new(region: get_region(context))
+
+  begin
+    ttl = Time.now.to_i + 120
+    client.put_item(
+      table_name: ENV['token_status_table'],
+      item: {
+        token_id: sid,
+        ttl: ttl
+      },
+      condition_expression: 'attribute_not_exists(token_id)'
+    )
+  rescue Aws::DynamoDB::Errors::ConditionalCheckFailedException
+    puts "TOKEN VALIDATION ERROR: #{TokenStatus::ALREADY_EXISTS} token_id: #{sid}"
+    return TokenStatus::VALID_HTTP
+  end
+
+  # Placeholder for when we'll actually vet each token
+  # to confirm that it should not be throttled.
+  client.update_item(
+    table_name: ENV['token_status_table'],
+    key: {token_id: sid},
+    update_expression: 'SET vetted = :v',
+    expression_attribute_values: {':v': true}
+  )
+
+  TokenStatus::VALID_HTTP
+end
+
+# ARN is of the format arn:aws:lambda:{region}:{account_id}:function:{lambda_name}
```

Comment:
I think I'm going to move some stuff into a `ThrottlingHelper` or something like that, this function is repeated and might be one of those things.

Reply:
Suresh left a note for me about this on one of my previous PRs, but I didn't get a chance to implement it. It sounds like there's an API that may help here. His notes: https://github.com/code-dot-org/javabuilder/pull/224#discussion_r821973035